### PR TITLE
Timelines and exhibits manual vertical positions

### PR DESCRIPTION
**Pull request by novikov-alexander**
Includes unnecessary datetime field and error messages fixes.
This update allows you to manually set the position of exhibits and timelines relative to the boundaries of their parents.

### CODE REVIEW

File: Source/Chronozoom.Entities/SQL/ManualMigration/CreateEntireSchema.sql [hunk 1]
```diff
@@ -116,11 +116,13 @@ CREATE TABLE [dbo].[Timelines]
     [ToYear]            [decimal](18, 7)        NOT NULL,
     [ForkNode]          [decimal](18, 2)        NOT NULL,
     [Height]            [decimal](18, 2)        NULL,
+	[OffsetY]           [decimal](5, 2)         NULL,
     [Timeline_Id]       [uniqueidentifier]      NULL,
     [Collection_Id]     [uniqueidentifier]      NULL,
     [SubtreeSize]       [int]                   NOT NULL        DEFAULT (0),
     [FromIsCirca]       [bit]                   NOT NULL        DEFAULT (0),
     [ToIsCirca]         [bit]                   NOT NULL        DEFAULT (0),
+    	[OffsetY]           decimal (18, 15)        NULL,
```
Comment (alterm4nn) [inline Source/Chronozoom.Entities/SQL/ManualMigration/CreateEntireSchema.sql]: Why are this added twice with different dimensions?

File: Source/Chronozoom.UI/scripts/authoring.js [hunk 2]
```diff
@@ -140,6 +140,49 @@ var CZ;
...
+                if (!((ael.top <= (tc.y))
+                    || (tc.y + tc.height <= ael.bottom))) {
+                    arranged = false;
+                    if (ael.top > locMax)
+                        locMax = ael.top;
+                }
+            });
+            
+            if (locMax > locMin)
+                return (locMax - locMin) * (1.001 / (Math.max(tc.offsetY,
```
Comment (alterm4nn) [inline Source/Chronozoom.UI/scripts/authoring.js]: What's the magic number 1.001?

Comment (novikov-alexander) [inline Source/Chronozoom.UI/scripts/authoring.js]: This is inverse proportion. I added epsilon just in case there are problems. But I think it might be replaced by 1.0

File: Source/Chronozoom.UI/ui/auth-edit-timeline-form.html [hunk 3]
```diff
@@ -30,13 +30,32 @@
...
+        <input class="cz-form-item-bottom-bound-offset cz-form-item-offset cz-input" type="text" maxlength="200">
+    </div>
 
     <br />
+    <br />
     <label class="cz-lightgray cz-form-label">Import background image from</label>
-    <div class="cz-form-medialist"></div>
- 
+    <div class="cz-form-medialist cz-medialist">
+        <div class="cz-medialist-item" title="bing" media-picker="bing">
```
Comment (alterm4nn) [inline Source/Chronozoom.UI/ui/auth-edit-timeline-form.html]: These items are being generated in JS counterpart during load, no need to add them here

File: Source/Chronozoom.Entities/Chronozoom.Entities.csproj [hunk 4]
```diff
@@ -171,6 +171,7 @@
     <Content Include="SQL\ManualMigration\201408130000000_MultipleCollections.sql" />
     <Content Include="SQL\ManualMigration\201409250000000_PubliclySearchableChange.sql" />
     <Content Include="SQL\ManualMigration\201502100000000_AddBackgroundUrlAspectRatio.sql" />
+    <Content Include="SQL\ManualMigration\201502170000000_AddYOffsetsAttributes.sql" />
```
Comment (alterm4nn) [inline Source/Chronozoom.Entities/Chronozoom.Entities.csproj]: Where is 201502160000000_AddExhibitYOffsetAttribute.sql that I saw in your database?

File: Source/Chronozoom.Entities/SQL/ManualMigration/201502170000000_AddYOffsetsAttributes.sql [hunk 5]
```diff
@@ -0,0 +1,16 @@
+﻿IF NOT EXISTS (SELECT * FROM sys.columns  WHERE [name] = N'OffsetY' AND [object_id] = OBJECT_ID(N'Exhibits'))
+BEGIN
+	ALTER TABLE [Timelines] ADD [OffsetY] DECIMAL (5, 2) NULL;
```
Comment (alterm4nn) [inline Source/Chronozoom.Entities/SQL/ManualMigration/201502170000000_AddYOffsetsAttributes.sql]: [Exhibits] then

File: Source/Chronozoom.UI/ui/auth-edit-timeline-form.html [hunk 6]
```diff
@@ -30,13 +30,32 @@
     <div class="cz-form-time-end" style="display: block"></div>
 
     <br />
-    <label class="cz-lightgray cz-form-label">End Time *</label>
-    <div class="cz-form-time-end" style="display: block"></div>
+    <label class="cz-lightgray cz-form-label cz-form-offset-label">
+        Top boundary (<span title="Distance to child timeline top measured in percents of parent timeline height" class="cz-form-percent-tooltip">%</span>)
```
Comment (alterm4nn) [inline Source/Chronozoom.UI/ui/auth-edit-timeline-form.html]: Bad description. Better to be "Timeline top position measured in percent of parent timeline height"

File: Source/Chronozoom.UI/ui/auth-edit-timeline-form.html [hunk 7]
```diff
@@ -30,13 +30,32 @@
...
+    </label>
+    <div class="cz-input-wrapper" style="display:inline-block; width:10%">
+        <input class="cz-form-item-top-bound-offset cz-form-item-offset cz-input" type="text" maxlength="200">
+    </div>
+
+    <br />
+    <label class="cz-lightgray cz-form-label cz-form-offset-label">
+        Bottom boundary (<span title="Distance to child timeline bottom measured in percents of parent timeline height" class="cz-form-percent-tooltip">%</span>)
```
Comment (alterm4nn) [inline Source/Chronozoom.UI/ui/auth-edit-timeline-form.html]: "Timeline bottom position measured in percent of parent timeline height"

File: Source/Chronozoom.Entities/SQL/ManualMigration/201502170000000_AddYOffsetsAttributes.sql (hunk 5 above)
Comment (novikov-alexander) [inline Source/Chronozoom.Entities/SQL/ManualMigration/201502170000000_AddYOffsetsAttributes.sql]: What do you mean?

File: Source/Chronozoom.UI/ui/auth-edit-timeline-form.html [hunk 8]
```diff
@@ -30,13 +30,25 @@
...
+        Bottom boundary (<span title="Timeline bottom position measured in percent of parent timeline height" class="cz-form-percent-tooltip">%</span>)
+    </label>
+    <div class="cz-input-wrapper" style="display:inline-block; width:10%">
+        <input class="cz-form-item-bottom-bound-offset cz-form-item-offset cz-input" type="text" maxlength="200">
+    </div>
 
     <br />
+    <br />
     <label class="cz-lightgray cz-form-label">Import background image from</label>
     <div class="cz-form-medialist"></div>
```
Comment (novikov-alexander) [inline Source/Chronozoom.UI/ui/auth-edit-timeline-form.html]: Should I leave this line?

File: Source/Chronozoom.Entities/Chronozoom.Entities.csproj (hunk 4 above)
Comment (novikov-alexander) [inline Source/Chronozoom.Entities/Chronozoom.Entities.csproj]: It were only in my own fork. I removed it before pull request creation and transferred the code to AddYOffsetsAttributes.sql. Is there any problem?

File: Source/Chronozoom.Entities/SQL/ManualMigration/201502170000000_AddYOffsetsAttributes.sql (hunk 5 above)
Comment (alterm4nn) [inline Source/Chronozoom.Entities/SQL/ManualMigration/201502170000000_AddYOffsetsAttributes.sql]: You try to add OffsetY to Timelines table twice. Copy&paste typo :-)

File: Source/Chronozoom.UI/ui/auth-edit-timeline-form.html (hunk 8 above)
Comment (alterm4nn) [inline Source/Chronozoom.UI/ui/auth-edit-timeline-form.html]: Yes.

File: Source/Chronozoom.UI/ui/auth-edit-exhibit-form.js [hunk 9]
```diff
@@ -258,13 +285,30 @@ var CZ;
...
+
+                    // If offset has changed, then we need to redraw layout.
+                    var isOffsetChanged = (Number(_this.offsetInput.val()) !== _this.exhibitCopy.offsetY);
+                    if (isOffsetChanged) {
+                        CZ.VCContent.clear(CZ.Common.vc.virtualCanvas("getLayerContent"));
+                        CZ.Common.reloadData().done(function () {
+                            _this.exhibit = CZ.Common.vc.virtualCanvas("findElement", _this.exhibit.id);
+                             delete self.timeline.animation;
+  
+ -                                    //Move to new created timeline
```
Comment (alterm4nn) [inline Source/Chronozoom.UI/ui/auth-edit-exhibit-form.js]: -/+ at first lines in code - garbage from unfinished merge with conflicts
